debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresStreamingChangeEventSource.java

```diff
@@ -92,7 +96,7 @@ public void execute(ChangeEventSourceContext context) throws InterruptedExceptio
         }
 
         try {
-            if (offsetContext.hasLastKnownPosition()) {
+            if (hasStartLsnStoredInContext) {
                 // start streaming from the last recorded position in the offset
                 final Long lsn = offsetContext.lastCompletelyProcessedLsn() != null ? offsetContext.lastCompletelyProcessedLsn() : offsetContext.lsn();
                 if (LOGGER.isDebugEnabled()) {
```

Comment:
Do you also need to change this a few lines down?
```
this.lastCompletelyProcessedLsn = offsetContext.lsn();
```

Reply:
@cjbooms I know about this one, I am still trying to find out how to propagate the value from `PostgreReplicationStream`. But thanks for reminder I forgot to mark this PR as WIP

Reply:
@cjbooms Should be fixed now